internal/mdns/dialable.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package mdns
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	manet "github.com/multiformats/go-multiaddr/net"
+)
+
+var (
+	ErrTCPListenAddressNotFound = errors.New("TCP listen address not found")
+)
+
+func getMDNSIPsAndPort(p2pHost Networker) (ips []net.IP, port uint16) {
```

Comment:
no tests for this package?

Reply:
I can add some, the reason I didn't was:

- the original mdns_legacy package we were using did not have any test (well it has one that was skipped and not working :cry:)
- this is a bit of a temporary fix, ideally we can upgrade to the newer mdns package, but this breaks all our stuff for now (see discussion above)
- rpc tests with multiple Gossamer nodes fail when that new mdns package is badly implemented

Reply:
Do we have an issue to track using the new mdns package? If we're not gonna add tests to this, we should ensure everything is working on staging before merging this.

Reply:
Created https://github.com/ChainSafe/gossamer/issues/2993
It's deploying now on staging.